test/tokens/TokenManager.js

```diff
@@ -0,0 +1,133 @@
+// Copyright © 2019 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global describe it */
+'use strict';
+
+const assert = require('assert');
+
+const TokenManager = require('../../lib/tokens/TokenManager');
+
+class TestTokenManager1 extends TokenManager {
+  constructor() {
+    super();
+    this._getTokenCallCount = 0;
+  }
+
+  _getToken(callback) {
+    this._getTokenCallCount += 1;
+    setTimeout(() => {
+      callback(null, { headers: { 'set-cookie': [ 'Max-Age=1' ] } });
+    }, 100);
+  }
+
+  // mock successful token renewal
+  get getTokenCallCount() {
+    return this._getTokenCallCount;
+  }
+}
+
+class TestTokenManager2 extends TokenManager {
```

Comment:
Rename to something like `FailedTestTokenManager` to show it's using a failed mocked token